### DISCUSSION TITLE
Error with beets installing the audible plugin

### DISCUSSION THREAD

**Discussion by s3781403** [seanap/beets-audible#8] [Q&A]
Hey!
I've come across this error when following the instructions:

There are two errors it seems installing / loading plugins, beetsplug.copyartifacts and the audible plugin.

Any help would be super appreciated! 

Thanks
`
sudo docker exec -it beets sh
root@53b073fe591f:/# beet --version
** error loading plugin copyartifacts:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/beets/plugins.py", line 268, in load_plugins
    namespace = __import__(modname, None, None)
ModuleNotFoundError: No module named 'beetsplug.copyartifacts'

** error loading plugin audible:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/beets/plugins.py", line 268, in load_plugins
    namespace = __import__(modname, None, None)
  File "/plugins/audible/audible.py", line 13, in <module>
    from natsort import os_sorted
ModuleNotFoundError: No module named 'natsort'

beets version 1.6.0
Python version 3.9.16
plugins: edit, fromfilename, plexupdate, scrub, web
`

**Comment by griev04** [comment]
Hello there,
it looks like some packages are no longer available in the base image. You may use an older image by editing your docker-compose.yml.
Just change the latest version
`image: lscr.io/linuxserver/beets:latest`
to a fixed one, like:
`image: lscr.io/linuxserver/beets:amd64-1.6.0-ls150`

**Comment by maxhoheiser** [comment]
Hi, yes there are two python packages missing: 
`natsort` and `markdownify` - manually installing them fixes the problem, thus adding them to the docker image would fix it too

**Reply by CedricPelchat** [reply]
What do you mean by adding the m? Adding them? lol Or is there a m parameter ?

**Comment by robflate** [comment]
This still seems to be a problem. Other than manually installing the missing packages, does anyone have a fix?

**Comment by kcross88** [comment]
For anyone that stumbles on this, it looks like there is already a script in place to fix the missing packages (install-deps.sh) but the scripts folder is mounted to the wrong folder in the container. 

See the Linuxserver.io documentation on it here: https://docs.linuxserver.io/general/container-customization/#custom-scripts

To fix, change the following in the docker compose volume section from:
      - ./scripts:/config/custom-cont-init.d
To:
      - ./scripts:/custom-cont-init.d:ro

To check if it worked, open an interactive shell in the container with `docker exec -it beets sh` and check the version `beet --version`. You shouldn't see any errors at this point. 

Just tried tagging a book and it seemed to work. Hopefully this helps someone!

**Reply by AP0LL0-19** [reply]
I get this error:
Error while requesting https://api.audnex.us/books/B0DQJKFB3P, attempt 1/3: status code 500, Internal Server Error
Error while requesting https://api.audnex.us/books/B0DQJKFB3P, attempt 2/3: status code 500, Internal Server Error
Error while requesting https://api.audnex.us/books/B0DQJKFB3P, attempt 3/3: status code 500, Internal Server Error
audible: Error while fetching book information from Audnex

as well as large traceback:

Traceback (most recent call last):
  File "/plugins/audible/audible.py", line 255, in get_albums
    return [self.get_album_info(p["asin"]) for p in products_without_unreleased_entries]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/plugins/audible/audible.py", line 264, in get_album_info
    (book, chapters) = get_book_info(asin)
                       ^^^^^^^^^^^^^^^^^^^
  File "/plugins/audible/api.py", line 24, in get_book_info
    book_response = json.loads(make_request(f"{AUDNEX_ENDPOINT}/books/{asin}"))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/plugins/audible/api.py", line 50, in make_request
    raise e
  File "/plugins/audible/api.py", line 42, in make_request
    with request.urlopen(req) as response:
         ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 215, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 521, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 630, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 559, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 492, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 639, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 500: Internal Server Error

I would love it if someone could help me as I don't know what to do with this

**Comment by eringuet** [comment]
You can: 
docker exec -it beets sh
pip install --no-cache-dir requests markdownify natsort beets-copyartifacts3
and it should work.

**Reply by Sol-Dawn** [reply]
Thanks this worked.
